sopel/plugins/handlers.py

```diff
@@ -265,6 +311,14 @@ def setup(self, bot):
             self._module.setup(bot)
 
     def has_setup(self):
+        """Tell if the plugin has a setup action.
+
+        :return: ``True`` if the plugin has a setup, ``False`` otherwise
+        :rtype: bool
+
+        The plugin has a setup action if its module has a ``setup`` attribute.
+        This attribute is expected to be a callable.
+        """
```

Comment:
I don't understand why this is duplicated. The `setup` attribute is always expected to be a callable, so why not just document the expectations in the `AbstractPluginHandler` instance of the method?

If there's good cause for having a duplicated docstring, update "action" to "routine" per above line-note suggestions.

Reply:
Because the `AbstractPluginHandler` doesn't assume you have a Python module for the plugin.

For instance, we could imagine a plugin defined as an object, or a class, that would, and a plugin handler for that type wouldn't need a module at all. In theory, you can already do that if you feel very adventurous, using `name = my_plugin:MyPluginObject`.

If `MyPluginObject` fakes just the right behaviours to pass as a Python module, that would work. I don't recommend it (yet), but it would 100% work.

So yeah, I prefer to duplicate, because it's an implementation detail, and not part of the interface.